src/operator/tensor/matrix_op-inl.h

```diff
@@ -471,10 +467,15 @@ void SliceDimOneCsrImpl(const TShape &begin, const TShape &end, const OpContext&
       MSHADOW_TYPE_SWITCH(in.dtype(), DType, {
         RType* in_indptr = in.aux_data(kIndPtr).dptr<RType>();
         RType* out_indptr = out.aux_data(kIndPtr).dptr<RType>();
-        SliceCsrIndPtrImpl<cpu, RType>(begin_row, end_row, ctx.run_ctx, in_indptr, out_indptr);
+        SliceCsrIndPtrImpl<xpu, RType>(begin_row, end_row, ctx.run_ctx, in_indptr, out_indptr);
 
-        // retrieve nnz (CPU implementation)
-        int nnz = out_indptr[indptr_len - 1];
+        Stream<xpu> *s = ctx.get_stream<xpu>();
+        // wait SliceCsrIndPtrImpl to complete
+        s->Wait();
```

Comment:
Since `nnz` is used immediately after `mshadow::Copy`. So `s->Wait()` is used here, to wait for the computation on `out_indptr` to complete.

Reply:
cudaMemcpy is done through default cuda stream and synchronous. So, it should wait to complete before it returns.

Reply:
Thanks for your comments. `stream->wait` is removed.